lib/elasticity/ebs_configuration.rb

```diff
@@ -0,0 +1,67 @@
+module Elasticity
+
+  class EBSConfiguration
+
+    def self.from_opts(ebs_opts = {})
+      if (ebs_opts[:ebs_size_in_gb].nil? || ebs_opts[:ebs_optimized].nil? || ebs_opts[:ebs_volume_type].nil? || ebs_opts[:ebs_number_of_volumes].nil?)
+        raise ArgumentError, "Missing EBS parameters. Passed: #{ebs_opts}"
+      end
+
+      ebs_opts[:ebs_size_in_gb] = ebs_opts[:ebs_size_in_gb].to_i
+      if ebs_opts[:ebs_iops]
+        ebs_opts[:ebs_iops] = ebs_opts[:ebs_iops].to_i
+      end
+
+      ebs_opts[:ebs_number_of_volumes] = ebs_opts[:ebs_number_of_volumes].to_i
+
+      if ebs_opts[:ebs_size_in_gb] <= 0
+        raise ArgumentError, "EBS Size must be at least 0 to add an EBS volume (#{ebs_opts[:ebs_size_in_gb]} requested)"
+      end
+
+      if ebs_opts[:ebs_number_of_volumes] < 1
+        raise ArgumentError, "Must have at least one EBS volume (#{ebs_opts[:ebs_number_of_volumes]} requested)"
+      end
+
+      if ebs_opts[:ebs_size_in_gb] >= 1024
```

Comment:
Where 1Tb limit is coming from?

Reply:
I don't know. This method was copied verbatim from the `InstanceGroup` class (at least that was my intent).

Reply:
Current limits are 16TB https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSVolumeTypes.html. Min size is per type, so we don't have a good support for that.

Reply:
I'm not sure if you're suggesting I change this. I'm going to default to leaving it the way it is unless I hear you'd like me to do otherwise.

Reply:
I'am suggesting to change max to 16TB. Min can stay at 1 GB.